src/MuTalk-Model/PercentRandomMutantSelection.class.st

```diff
@@ -0,0 +1,42 @@
+Class {
+	#name : #PercentRandomMutantSelection,
+	#superclass : #RandomMutantSelection,
+	#instVars : [
+		'percentageOfMutants'
+	],
+	#category : #'MuTalk-Model-Mutations generation strategies'
+}
+
+{ #category : #accessing }
+PercentRandomMutantSelection >> percentageOfMutants [
+
+	^ percentageOfMutants
+]
+
+{ #category : #accessing }
+PercentRandomMutantSelection >> percentageOfMutants: aPercentage [
+
+	(aPercentage < 0 or: [ aPercentage > 100 ]) ifTrue: [
+		Error signal: 'Percantage should be between 0 and 100' ].
+	percentageOfMutants := aPercentage
+]
+
+{ #category : #accessing }
+PercentRandomMutantSelection >> selectMutantsFrom: aCollection [
+
+	| size index newColl random numberOfMutants |
+	size := aCollection size.
+	newColl := aCollection copyEmpty.
+	random := Random new.
+	numberOfMutants := size * percentageOfMutants / 100.
```

Comment:
Cool, so you transformed a percentage into an absolute number of mutants.
Now you can reuse the code from the other class!! How could you do it? :)